nomos-services/data-availability/dispersal/src/adapters/mempool/mod.rs

```diff
@@ -1,10 +1,28 @@
 pub mod kzgrs;
 
+use std::error::Error;
+
+use nomos_mempool::backend::MempoolError;
 use overwatch::{
     services::{relay::OutboundRelay, ServiceData},
     DynError,
 };
 
+#[derive(Debug)]
+pub enum DaMempoolAdapterError {
+    Mempool(MempoolError),
+    Other(DynError),
+}
+
+impl DaMempoolAdapterError {
+    fn to_other<E>(e: E) -> Self
+    where
+        E: Error + Send + Sync + 'static,
+    {
+        Self::Other(Box::new(e) as DynError)
+    }
+}
+
```

Comment:
I think it would be better to implement `impl<E: Error + Send + Sync + 'static> From<E> for DaMempoolAdapterError`. It can just be added.